### PR TITLE
fix: resolve opts when no-config & fix vulns

### PR DESCRIPTION
**Pull request by evenstensberg**
**What kind of change does this PR introduce?**
Closes https://github.com/webpack/webpack-cli/issues/1023
**Did you add tests for your changes?**
Yes
**If relevant, did you update the documentation?**
N/A
**Summary**
Dupes out of resolving options if there are none to throw an error

**Does this PR introduce a breaking change?**
No
**Other information**
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -142,7 +142,7 @@
     "babel-preset-env": "1.7.0",
     "babel-preset-jest": "24.6.0",
     "codecov": "3.5.0",
-    "commitizen": "3.1.1",
+    "commitizen": "^4.0.3",
```
Comment (ematipico) [inline package.json]:
```suggestion
    "commitizen": "4.0.3",
```

File: package.json [hunk 2]
```diff
@@ -131,7 +131,7 @@
   "devDependencies": {
     "@babel/preset-env": "7.4.5",
     "@babel/register": "7.4.4",
-    "@commitlint/cli": "8.0.0",
+    "@commitlint/cli": "^8.1.0",
```
Comment (ematipico) [inline package.json]:
```suggestion
    "@commitlint/cli": "8.1.0",
```

File: bin/utils/convert-argv.js [hunk 3]
```diff
@@ -128,28 +128,32 @@ module.exports = function(...args) {
...
-			process.exit(-1); // eslint-disable-line
+		if (options) {
+			const webpackConfigurationValidationErrors = validateSchema(webpackConfigurationSchema, options);
+			if (webpackConfigurationValidationErrors.length) {
+				const error = new WebpackOptionsValidationError(webpackConfigurationValidationErrors);
+				console.error(error.message, `\nReceived: ${typeof options} : ${JSON.stringify(options, null, 2)}`);
+				process.exit(-1); // eslint-disable-line
+			}
+		} else {
+			options = {};
```
Comment (ematipico) [inline bin/utils/convert-argv.js]: Nothing wrong with this code, but I usually avoid mutations like this one.

File: bin/utils/convert-argv.js [hunk 4]
```diff
@@ -128,28 +128,32 @@ module.exports = function(...args) {
...
 		}
 
 		// process Promise
-		if (typeof options.then === "function") {
+		if (options && typeof options.then === "function") {
```
Comment (ematipico) [inline bin/utils/convert-argv.js]: Now that you have the if/else, `options` will never be undefined right? So this additional check is not needed

Comment (evenstensberg) [reply bin/utils/convert-argv.js]: checks for `options.then` object prop so will throw. We're setting props on the object later these checks, I think we "assumed" we either found or did not find the props before validating args that aren't protected by 0CJS (entry and output)

File: bin/utils/convert-argv.js (hunk 3 above)
Comment (evenstensberg) [reply bin/utils/convert-argv.js]: Yea, agreed. I think we would have needed a larger refactor to process args without validating options, as explained a bit further down

File: bin/utils/convert-argv.js (hunk 4 above)
Comment (ematipico) [reply bin/utils/convert-argv.js]: I'm not talking about .then, I'm talking about `options &&`. It's redundant. You already assigned it before with `options = {}`. 

So or you keep this check and remove the assignment or keep the assignment and remove the check.

Comment (evenstensberg) [reply bin/utils/convert-argv.js]: Thanks for pointing it out, changed now!

**Pull request merged**